### ISSUE TITLE
mAP is always zero on COCO dataset

### ISSUE THREAD

**Issue by IaroslavS** [xuannianz/keras-CenterNet#19]
Hi, everyone.
I wonder if prediction model is being trained while training model is being trained ?
I think there's an error in file eval/coco.py line 111 :
def __init__(self, generator, tensorboard=None, threshold=0.05):
There should be extra argument - "prediction_model" since in train.py line 88 looks like this 
evaluation = CocoEval(validation_generator, prediction_model, tensorboard=tensorboard_callback)
and this CocoEval gives argument "prediction_model". Thus line 111 of eval/coco.py should be like this:
def __init__(self, generator, model, tensorboard=None, threshold=0.05):

And there's an error occurs in line 47 of eval/coco.py because it tries to use training model instead of prediction model. Though we send argument prediction_model to CocoEval, never the less this variable is modified somehow (I didn't find where exactly) in other packages of library. I think, variable self.model should be changes to another unique, for example, self.prediction_model. And it will become like this:

class CocoEval(keras.callbacks.Callback):
    """ Performs COCO evaluation on each epoch.
    """
------------------------------------------------------------------------------------------------------
    def __init__(self, generator, model, tensorboard=None, threshold=0.05):
        """ CocoEval callback intializer.

        Args
            generator   : The generator used for creating validation data.
            tensorboard : If given, the results will be written to tensorboard.
            threshold   : The score threshold to use.
        """
        self.generator = generator
        self.threshold = threshold
        self.tensorboard = tensorboard
        self.prediction_model = model
        super(CocoEval, self).__init__()
-------------------------------------------------------------------------------------------------------------------
And even if you'll correct all the above mistakes I'm not sure that training model for COCO dataset will be correct, because right now my trained model on my custom COCO dataset has zero detections on all images, though val loss fell from about 200 to 6.

**Comment by klyjm** [comment]
I have the same error when I use csv, if the image size is small than the args.input_size, the mAP will always zero, I don't know why, and I haven't find any error in pascal evaluate······

**Comment by Gateway2745** [comment]
I also getting the AP scores as 0 on all classes on my custom dataset. Any thoughts on why this is happening ?

**Comment by IaroslavS** [comment]
No, I gave up with CenterNet.

Отправлено с iPhone
